codex-ui/src/vue/components/context-menu/ContextMenu.vue

```diff
@@ -41,10 +48,56 @@ withDefaults(
      * Array of items for context menu
      */
     items: Item[];
+
+    /**
+     * Message what occurs as a result of the search
+     */
+    message: string;
   }>(),
   { showSearch: false }
 );
 
+/**
+ * User entered word for search
+ */
+const searchTerm = ref('');
+
+/**
+ * Returns the list of menu context items found during the search
+ */
+const filteredItems = computed(() => {
+  if (searchTerm.value === '') {
+    return props.items;
+  }
+
+  const searchedItems = props.items.filter((item) => {
+    if (item.type === 'message') {
+      return false;
+    } else if (item.type === 'separator') {
+      return true;
+    } else {
+      return item.title.toLowerCase().includes(
+        searchTerm.value.toLowerCase());
+    }
+  });
+
+  if (searchedItems.length > 0) {
+    return searchedItems[0].type === 'separator'
+      ? searchedItems.slice(1)
+      : searchedItems;
+  }
+
+  return searchedItems;
```

Comment:
(*) we have this case only when `serchedItems.lenght === 0`, so we can just return message item here